File: 326-PowerOfThree/326-PowerOfThree.java
// Last updated: 6/10/2026, 7:25:11 PM
class Solution {
    public boolean isPowerOfThree(int n) {
        if (n <= 0) return false;
        while (n % 3 == 0) {
            n /= 3;
        }
        return (n == 1);
    }
}

File: 326-PowerOfThree/326-PowerOfThree.cpp
// Last updated: 6/10/2026, 7:25:17 PM
class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n==0){
            return false;
        }
        else if(n==1){
            return true;
        }
        return n % 3==0 && isPowerOfThree(n/3); 
    }
};